braidpool-primitives/src/bead/mod.rs

```diff
@@ -102,34 +226,11 @@ impl Bead {
         // The bead might get pruned later, so we can't give a shared reference!
         self.committed_metadata.parents.clone()
     }
-
-    // #[inline]
-    // pub fn get_children(&self) -> Children {
-    //     self.children.borrow().iter().cloned().collect()
-    // }
 }
 
 impl Bead {
-    // All pub(crate) function definitions go here!
-    #[inline]
-    pub(crate) fn add_child(&self, child_bead_hash: BeadHash) {
-        // TODO: While Implementing the DB, we also need to update the corresponding DB Entry!
-        // self.children.borrow_mut().insert(child_bead_hash);
-        unimplemented!();
-    }
-
     pub fn get_bead_hash(&self) -> BlockHash {
-        // Needs to be changed
-        // Ideally, the hash should be equivalent to the block hash
-        let serialized_bead_str = serde_json::to_string(self).unwrap_or_else(|error| {
-            panic!(
-                "An error occurred while serializing the bead string: {:?}",
-                error
-            )
-        });
-        let mut serialized_bytes = [0u8; 32];
-        hex::decode_to_slice(serialized_bead_str, &mut serialized_bytes).unwrap();
-        BlockHash::from_byte_array(serialized_bytes)
+        self.block_header.block_hash()
     }
 }
 
```

Comment:
This would be a lot easier if we were carrying around Bead objects instead of BeadHash objects.